processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -970,6 +970,22 @@ public GroupByQuery build()
     }
   }
 
+  @Override
+  public QuerySegmentSpec getQuerySegmentSpecForLookUp()
+  {
+    return getInnerMostQuerySegmentSpec(this);
+  }
+
+  private static QuerySegmentSpec getInnerMostQuerySegmentSpec(GroupByQuery query)
```

Comment:
This method looks like it's not specific to GroupByQuerys. So instead would it make sense to add this logic to getRunner rather than having a new getQuerySegmentSpecForLookUp method? `getDataSource` exists on all query types and I think it'd always make sense to traverse down through QueryDataSources.

Reply:
I was under the impression that nested queries are only supported for group by queries. If that it isn't the case, yes, this code can be moved to getRunner() method in the BaseQuery class.

Reply:
They are currently only supported for groupBy queries. I was thinking it would be harmless to add the code to BaseQuery though, and simplify things a bit.

Reply:
I would say move it there to avoid the need to add the extra methods.

Reply:
Thanks for the review. I have made suggested changes in the latest commit.